### DISCUSSION TITLE
Save Game Formats

### DISCUSSION THREAD

**Discussion by stephen-hill** [MattFiler/LegendPlugin#183] [General]
Hi

I've started work on trying to reverse engineer the same game formats: CHS and PROGRESS.

So far I've made the following assumptions:

- Each file starts for "IAAC"
- Then 4 bytes which contain a version number. (12 for CHSV and 4 for PROGRESS)
- Then 10 bytes which appear to be a checksum/hash - however, a 10 byte hash is unusual.
- PROGRESS then appears to be serialization of memory.

That as far as I've got. My first goal will be to figure out if there is a hash at the top of the files, and if so, how it is generated.

Anyway, the reason I am writing all this is I am wondering if anyone has more information about the save game formats?

Cheers
Stephen

**Comment by MattFiler** [comment]
Hey, I did briefly look at the PROGRESSION.AIS file a few months ago ([here](https://github.com/OpenCAGE/CathodeLib/blob/1704f3ea0862a6fb08808e95dce214be466fbe19/CathodeLib/Scripts/File%20Handlers/Saves/CathodeProgression.cs)), but haven't explored saves all that much. I figured out where the config values were by changing menu options and saving. Figuring out the whole save format would be a super interesting task (and one that may be helped by the decompilation of the recent iOS release of the game, which contains all symbols), however isn't currently something on my roadmap.

**Comment by MattFiler** [comment]
I took a poke into the AIS save format today and made a little bit of progress. Essentially the file is broken down into a bunch of sections related to certain game systems, each with a header and subheaders. It stores info for a ton of Commands states, as well as all NPCs, collectables, inventory items, etc. I've got a few headers semi-reliably loading [here](https://github.com/OpenCAGE/CathodeLib/blob/staging/CathodeLib/Scripts/CATHODE/MissionSave.cs), but there are a lot more to add.
